_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,194 @@
+
+The [faas-cli](https://docs.openfaas.com/cli/install/) binary along with populated `OPENFAAS_URL` environmental variable pointing at your gateway.
+
+## Clone the Kafka Connector
+
+Pull the `kafka-connector` code from github:
+
+```bash
+$ git clone https://github.com/openfaas-incubator/kafka-connector
+```
+
+Inside `kafka-connector` folder navigate to `yaml/kubernetes`:
+
+```bash
+$ cd kafka-connector/yaml/kubernetes
+```
+
+Here we can see the Kafka Zookeeper and Broker which we will be using to send payloads to our functions.
+
+## Deploy the components with configuration
+
+In order to deploy the Zookeeper and Broker apply the kubernetes files while inside the kubernetes folder from above:
+
+```bash
+$ kubectl apply -f kafka-broker-dep.yml,kafka-broker-svc.yml,zookeeper-dep.yaml,zookeeper-svc.yaml
+```
+
+It is important to note that those files deploy the Zookeeper and Broker in the same namespace as the OpenFaaS Gateway.
+
+We will use charts to deploy the Kafka Connector. If you don't have the OpenFaaS charts repository, then add it with this command:
+
+```bash
+$ helm repo add openfaas https://openfaas.github.io/faas-netes/
+```
+
+Install Kafka Connector with default values:
+
+```bash
+$ helm upgrade kafka-connector openfaas/kafka-connector \
+    --install \
+    --namespace openfaas \
+    --set topics="faas-request\," \
+    --set broker_host="kafka" \
+    --set print_response_body="true"
+```
+
+Here it is important to note that our Broker resides in service called `kafka` in the same namespace as our Gateway and Kafka Connector service. If you have Broker service in another namespace, again called `kafka`, just append the namespace like so `kafka.<your_namespace>` to the `broker_host` environmental variable so the Connector can discover it. The default kafka port `9092` is appended to the service name by default.
+
+Also if you would like to subscribe the connector and your functions to more than one topic, list them in the `topics` variable by separating them with comma and re-run the command above.
+
+Congratulations now you have the whole stack deployed:
+* Zookeeper
+* Broker Host with Producer
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas | grep -E 'kafka|zookeeper|producer'
+kafka-broker-544fbccd48-p6pgr     1/1     Running            3          71m
+kafka-connector-5d9f447f5-5drv7   1/1     Running            0          69m
+zookeeper-699f568f6f-6b4n2        1/1     Running            0          71m
+```
+
+## Subscribe function
+
+In order to consume topics via the connector we simply need to apply an annotation with a key of `topic` and the chosen value used in the `connector-dep.yml` - in our case it was `faas-request`.
+
+Here it is important to note that if you would like your function to subscribe to multiple topics, you just need to list them as we did in the connector configuration.
```

Comment:
The connector can bind to multiple topics, but functions cannot yet.

Reply:
Should I remove this? Or we will merge it before the blog is out ? I was hoping to get it in the project before publishing the blog

Reply:
Removed the reference